src/dfilemaker/dfilemaker.c

```diff
@@ -580,25 +653,106 @@ int main(int narg, char** arg)
+        *  shortopts below are followed by a colon if they take an argument  
+        *---------------------------------------------------------------------*/        
+        c=getopt_long(narg,arg,"i:f:d:n:r:s:w:vh",long_options,&longind);
+        if (c <= 0) break;
+        minterm=(char*)MFU_MALLOC(10*sizeof(char));
+        maxterm=(char*)MFU_MALLOC(10*sizeof(char));
+        switch (c) {
+           case 'i':
+             jseed = atoi(optarg);
+             if (jseed) iseed=jseed;
+             break;
+           case 'f':
+             for (kft=0;kft<4;kft++) if (strcmp(optarg,filltype[kft])==0) break;
+             if (kft==4 && rank==0)
+             {
+                printf("%s not a fill option\n",optarg);
+                MPI_Finalize();  
+             }
+             break;
+           case 'd': 
+             // range for nlevels
+             mmparse(optarg,&minterm,&maxterm);
+             depmin=atoi(minterm);
+             depmax=atoi(maxterm);
+             break;
+           case 'n':
+              // range for ntotal 
+              mmparse(optarg,&minterm,&maxterm);
+              nmin=atoi(minterm);
+              nmax=atoi(maxterm);
+              break;
+            case 'r':
+              ratio = atof(optarg);
+              printf("ratio = %f\n",ratio);
+              break;
+            case 's': 
+              // range for maxflen
+              mmparse(optarg,&minterm,&maxterm);
+
+              /* read file size */
+              if (mfu_abtoull(minterm, &sizeminl) != MFU_SUCCESS)
+              {
+                if (rank == 0)
+                {
+                  printf("Could not interpret %s as file size\n", minterm);
+                  fflush(stdout);
+                }
+                mfu_finalize();
+                MPI_Finalize();
+                return 1;
+              }
+              sizemin=(int)sizeminl;
+              if (mfu_abtoull(maxterm, &sizemaxl) != MFU_SUCCESS)
+              {
+                if (rank == 0)
+                {
+                  printf("Could not interpret %s as file size\n", maxterm);
+                  fflush(stdout);
+                }
+                mfu_finalize();
+                MPI_Finalize();
+                return 1;
+              }
+              sizemax=(int)sizemaxl;
+              break;
+            case 'w': 
+              mmparse(optarg,&minterm,&maxterm);
+              widmin=atoi(minterm);
+              widmax=atoi(maxterm);
+              break;
+            case 'v':
+              if (rank == 0) printf("verbose is on\n");
+              break;
+            case 'h':
+              if (rank == 0) print_usage();
+              break;
+            default: 
+              break;
+        }
+     };
+     srand(iseed);
+     if (nmax > 0) ntotal=nmin+rand()%(1+nmax-nmin);
+     if (depmax > 0) nlevels=depmin+rand()%(1+depmax-depmin);
+     if (sizemax > 0) maxflen=sizemin+rand()%(1+sizemax-sizemin);
```

Comment:
Let's use braces and indentation here, too.

Reply:
Addressed in #604